en/tutorials-and-examples/cms/authentication.rst

```diff
@@ -48,7 +48,7 @@ add the following::
     <?php
     namespace App\Model\Entity;
 
-    use Authentication\PasswordHasher\DefaultPasswordHasher; // Add this line
```

Comment:
But that line is absolutely correct, this is what it refers to: https://github.com/cakephp/authentication/blob/2.3.0/src/PasswordHasher/DefaultPasswordHasher.php

It sounds more like you've possibly made a mistake and not included the authentication plugin (correctly)?

Reply:
I agree, this is the correct `use` statement for Authentication plugin.

Reply:
> It sounds more like you've possibly made a mistake and not included the authentication plugin (correctly)?

Yeah, that could be an option. But I just imported the `DefaultPasswordHasher` statement  like this:

_User.php_ 
```
<?php
declare(strict_types=1);

namespace App\Model\Entity;

use Authentication\PasswordHasher\DefaultPasswordHasher;
use Cake\ORM\Entity;

... 
```

There are another line that should I include? Can you help me to understand what I'm doing wrong?

Reply:
The file looks fine, what I ment to refer to are the actual classes/files of the authentication plugin.

You need to require the plugin via composer [as shown here](https://book.cakephp.org/4/en/tutorials-and-examples/cms/authentication.html#installing-authentication-plugin), have you successfully done that?

Reply:
🤦 thank you ! @ndm2 ... i didn't realized that this was an external plugin. That worked !! 👏